src/jshint.js

```diff
@@ -2218,6 +2217,19 @@ var JSHINT = (function() {
     return this;
   }));
   prefix("new", function() {
+    var mp = metaProperty("target");
+    if (mp) {
+      if (mp.value === "target") {
+        if (!state.inESNext(true)) {
+          warning("W119", state.tokens.prev, "new.target");
+        }
+        if (!state.inClassBody) {
```

Comment:
the arrow function is simply ignored. So, you need to know, am I in a non-arrow function (somewhere on the stack). I guess you might need a new state property? but also this is exactly the same behaviour as `arguments` - can't remember how that is done though, it might be done by adding a variable to the scope, which is clearly not an option.